Add support for pytest>=0.5.0 to nbdime_reporter.py

kanderso-nrel opened the pull request:
`nbdime_reporter.py` errors for pytest>0.5.0 because pytest changed the error codes to be housed in an Enum (see
https://github.com/pytest-dev/pytest/commit/2b92fee1c315469c3f257673a508e51a1f1a6230#diff-7191acb1a6aaab2f77b6e53ce837e348):


```
(rdtools-dev) C:\Users\KANDERSO\projects\rdtools>pytest --nbval --nbdime
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\_pytest\main.py", line 265, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\_pytest\config\__init__.py", line 982, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\pluggy\hooks.py", line 308, in call_historic
INTERNALERROR>     res = self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\pluggy\manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\pluggy\manager.py", line 87, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\pluggy\callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\pluggy\callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\pluggy\callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\nbval\plugin.py", line 104, in pytest_configure
INTERNALERROR>     from .nbdime_reporter import NbdimeReporter
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\_pytest\assertion\rewrite.py", line 170, in exec_module
INTERNALERROR>     exec(co, module.__dict__)
INTERNALERROR>   File "c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\nbval\nbdime_reporter.py", line 10, in <module>
INTERNALERROR>     from _pytest.main import EXIT_OK, EXIT_TESTSFAILED, EXIT_INTERRUPTED, \
INTERNALERROR> ImportError: cannot import name 'EXIT_OK' from '_pytest.main' (c:\users\kanderso\software\anaconda3\envs\rdtools-dev\lib\site-packages\_pytest\main.py)
```

This PR gets it working for me locally with pytest 4.6.11, 5.4.1, and 6.2.3.  I would be happy to push up other changes, but I'm not sure what's needed (changelog?  tests?) -- let me know.

kanderso-nrel commented:
Can't say I understand the appveyor failure, but as the failures don't involve the `--nbdime` flag I think they are unrelated to this PR?

kanderso-nrel commented:
Just checking in -- I'm still happy to add to this PR if it's of interest 😁

takluyver commented:
Sorry for the lack of attention to this.

Close/reopen to re-run tests with changes in master.

takluyver commented:
Tests are passing again. I think it would be good to figure out some basic check of the nbdime integration in the tests - as things stand, it's evidently not even importing that module in the tests, or this would have been noticed much sooner.

kanderso-nrel commented:
Thanks @takluyver, I added a test.  Let me know if it's along the lines of what you were thinking.  I added pytest-mock as a dependency, hopefully that's okay -- otherwise we'll have to find some other way to prevent the nbdime server from spawning and stalling the pytest process.

takluyver commented:
Thanks, that looks good to me.
